mapfishapp/src/main/webapp/app/js/GEOR_Querier.js

```diff
@@ -0,0 +1,330 @@
+ * @include Styler/widgets/FilterBuilder.js
+ * @include GEOR_waiter.js
+ * @include GEOR_ows.js
+ * @include GEOR_util.js
+ */
+
+/* globals Ext, GeoExt, OpenLayers, GEOR */
+
+Ext.namespace("GEOR");
+
+
+GEOR.Querier = Ext.extend(Ext.Window, {
+
+    layout: "fit",
+    closeAction: "close",
+    border: false,
+ 
+    /**
+     * Property: filterbuilder
+     * {String} The local Styler.FilterBuilder instance
+     */
+    filterbuilder: null,
+
+    /**
+     * Property: pseudoRecord
+     * {Object} Object with typeName & owsURL properties
+     */
+    pseudoRecord: null,
+
+    /**
+     * Property: geometryName
+     * {String} The geometry column name
+     */
+    geometryName: null,
+
+    /**
+     * Property: record
+     * {Ext.data.Record} The WMS or WFS layer record
+     */
+    record: null,
+
+    /**
+     * Property: map
+     * {OpenLayers.Map} The map instance.
+     */
+    map: null,
+
+    /**
+     * Property: layer
+     * {OpenLayers.Layer.Vector} The layer on which geometries can be drawn
+     */
+    layer: null,
+
+    /**
+     * Property: attributeStore
+     * {GeoExt.data.AttributeStore} 
+     */
+    attributeStore: null,
+
+    /**
+     * Property: layerFields
+     * {Array} an array of fields for the current layer
+     * It is extracted from the WFS DescribeFeatureType operation
+     */  
+    layerFields: null,
+
+    /*
+     * Method: initComponent.
+     * Overridden constructor. Set up widgets and lay them out
+     */
+    initComponent: function() {
+        var r = this.record,
+            type = r.get("type"),
+            isWFS = type === "WFS",
+            layer = r.get("layer"), 
+            name = r.get("title") || layer.name || "";
+
+        this.title = OpenLayers.i18n("Request on NAME", {
+            "NAME": name
+        });
+
+        this.pseudoRecord = {
+            typeName: isWFS ? 
+                r.get("WFS_typeName") : r.get("name"),
+            owsURL: isWFS ? 
+                layer.protocol.url : r.get("WFS_URL")
+        };
+
+        GEOR.waiter.show();
+        // get layer model through WFS DescribeFeatureType:
+        this.attributeStore = GEOR.ows.WFSDescribeFeatureType(this.pseudoRecord, {
+            extractFeatureNS: true,
+            success: function() {
+                // we list all fields, including the geometry
+                this.layerFields = this.attributeStore.collect("name");
+                // we get the geometry column name, and remove the corresponding record from store
+                var idx = this.attributeStore.find("type", GEOR.ows.matchGeomProperty);
+                if (idx > -1) {
+                    // we have a geometry
+                    var r = this.attributeStore.getAt(idx);
+                    this.geometryName = r.get("name");
+                    this.attributeStore.remove(r);
+                } else {
+                    GEOR.util.infoDialog({
+                        msg: tr("querier.layer.no.geom")
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["tr" is not defined.](https://www.codacy.com/app/georchestra/georchestra/file/2338282099/issues/source?bid=3200181&fileBranchId=3209168#l147)